libs/py-core/tests/test_config.py

```
@@ -0,0 +1,58 @@
+        assert settings.app_name == "test-app"
+        assert settings.environment == "production"
+        assert settings.debug is True
+        assert settings.log_level == "DEBUG"
+
+    def test_rejects_invalid_environment(
+        self, monkeypatch: pytest.MonkeyPatch
+    ) -> None:
+        """Settings should reject environment values not in the allowed list."""
+        monkeypatch.setenv("ENVIRONMENT", "invalid")
+
+        with pytest.raises(ValidationError):
```

Ruff (0.14.14)

[error] Undefined name `ValidationError`

(F821)